### ISSUE TITLE
Xcode Archive fails with "No such file or directory (os error 2)" in Bundle React Native code and images

### ISSUE THREAD

**Issue by Off2Race** [getsentry/sentry-react-native#3305]
**OS:**
- [X]  MacOS

**Platform:**
- [X]  iOS

**SDK:**
- [X]  `@sentry/react-native`

**SDK version:** 5.9.2

**`react-native` version:** 0.67.5

**Are you using Expo?**
- [X]  No

**Are you using sentry.io or on-premise?**
- [X]  sentry.io (SaaS)

---
**I have following issue:**

With @sentry/react-native 5.9.2 and @sentry/cli 2.20.7, the Xcode Archive command succeeds without issue.

If I upgrade @sentry/cli to 2.21.1 (with no other changes), I get the error shown below during the Bundle React Native code and images phase of the build.

```
info Writing bundle output to:, /Users/mikeloos/Library/Developer/Xcode/DerivedData/Questing-clqhjnzhujzcdfexariyjgtaqqsg/Build/Intermediates.noindex/ArchiveIntermediates/Questing/BuildProductsPath/Release-iphoneos/main.jsbundle
info Writing sourcemap output to:, /Users/mikeloos/Library/Developer/Xcode/DerivedData/Questing-clqhjnzhujzcdfexariyjgtaqqsg/Build/Intermediates.noindex/ArchiveIntermediates/Questing/IntermediateBuildFilesPath/Questing.build/Release-iphoneos/Questing.build/DerivedSources/main.jsbundle.map
info Done writing bundle output
info Done writing sourcemap output
info Copying 49 asset files
info Done copying assets
+ [[ true != true ]]
+ EXTRA_COMPILER_ARGS=
+ [[ false == true ]]
+ EXTRA_COMPILER_ARGS=-O
+ [[ '' == true ]]
+ /Users/mikeloos/Code/questing/client/questing/mobile/node_modules/@sentry/cli/sentry-cli -emit-binary -O -out /Users/mikeloos/Library/Developer/Xcode/DerivedData/Questing-clqhjnzhujzcdfexariyjgtaqqsg/Build/Intermediates.noindex/ArchiveIntermediates/Questing/BuildProductsPath/Release-iphoneos/Questing.app/main.jsbundle /Users/mikeloos/Library/Developer/Xcode/DerivedData/Questing-clqhjnzhujzcdfexariyjgtaqqsg/Build/Intermediates.noindex/ArchiveIntermediates/Questing/BuildProductsPath/Release-iphoneos/main.jsbundle
error: No such file or directory (os error 2)
```
I tried (unsuccessfully) to set the log level higher to get more information but nothing additional was displayed.

The script for the Bundle React Native code and images phase is very standard. See below.
```
export SENTRY_PROPERTIES=sentry.properties
export EXTRA_PACKAGER_ARGS="--sourcemap-output $DERIVED_FILE_DIR/main.jsbundle.map"

../node_modules/@sentry/cli/bin/sentry-cli react-native xcode
../node_modules/react-native/scripts/react-native-xcode.sh
../node_modules/@sentry/react-native/scripts/collect-modules.sh
```
If I downgrade the CLI back to 2.20.7, this issue goes away.

If I add the following line to the above script, the issue also goes away.

```
export USE_HERMES=0
```
This might seem like an odd thing to try but I got the idea while poking around the Xcode log. The app does not use Hermes, so I figured that setting this flag would be otherwise harmless.

**Steps to reproduce:**
Upgrade the CLI from 2.20.7 to 2.21.1. See more details above.

**Actual result:**
Xcode archive fails

**Expected result:**
Xcode archive succeeds

**Comment by krystofwoldrich** [comment]
Thank you for the message,
fix is already done here -> https://github.com/getsentry/sentry-cli/pull/1754

Will be released soon.

**Comment by Off2Race** [comment]
Hi, @krystofwoldrich – Is this issue resolved in sentry-react-native 5.11.0? It appears that the root fix was included in sentry-cli 2.21.2, which is a dependency of 5.11.0 but I wanted to ask to make sure before upgrading.

If I'm correct, perhaps you might add a note to the 5.11.0 release notes? :)

**Comment by krystofwoldrich** [comment]
@Off2Race Yes, this was fixed in 5.11 and newer.

We only include the information about CLI version bump as we never released RN SDK with the mentioned 2.21.1.

This is the related CLI changelog https://github.com/getsentry/sentry-cli/releases/tag/2.21.2
